app/models/custom_field.rb

```diff
@@ -70,6 +72,10 @@ def self.loan_questions(field_set = nil)
       { internal_name: field_set })
   end
 
+  def self.sort_by_required(loan)
+    all.sort_by { |i| [i.required_for?(loan) ? 0 : 1, i.position] }
```

Comment:
It could just as well be. I prefer class methods to scopes because they have exactly the same effect and I don't have to remember another syntax. I read that recommended somewhere—I think in Rails' own documentation.